QuizWhois.Api/Controllers/AuthController.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Google.Apis.Auth;
+using Google.Apis.Auth.AspNetCore3;
+using Google.Apis.Auth.OAuth2;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Newtonsoft.Json;
+using QuizWhois.Common.Models;
+using QuizWhois.Domain.Entity;
+
+namespace QuizWhois.Api.Controllers
+{
+    [ApiController]
+    [Route("auth")]
+    public class AuthController : Controller
+    {        
+        [Authorize]
+        [HttpPost("logout")]
+        public async Task<IActionResult> LogOut()
+        {
+            await HttpContext.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
+            return Ok("Success LogOut");
+        }
+
+        [Authorize]
+        [HttpGet("scopes")]
+        public async Task<ActionResult<List<string>>> ScopeListing([FromServices] IGoogleAuthProvider auth)
+        {
+            IEnumerable<string> currentScopes = await auth.GetCurrentScopesAsync();
+            return currentScopes.ToList();
+        }
+
+        [Authorize]
+        [HttpGet("tokens")]
+        public async Task<ActionResult<string>> ShowTokens()
+        {
+            // The user is already authenticated, so this call won't trigger authentication.
+            // But it allows us to access the AuthenticateResult object that we can inspect
+            // to obtain token related values.
+            AuthenticateResult auth = await HttpContext.AuthenticateAsync();
+            string idToken = auth.Properties.GetTokenValue(OpenIdConnectParameterNames.IdToken);
+            string idTokenValid, idTokenIssued, idTokenExpires;
+            try
+            {
+                var payload = await GoogleJsonWebSignature.ValidateAsync(idToken);
+                idTokenValid = "true";
+                idTokenIssued = new DateTime(1970, 1, 1).AddSeconds(payload.IssuedAtTimeSeconds.Value).ToString();
```

Comment:
wit, but doesn't DateTimeOffset.ToUnixTimeSeconds do the same?

Reply:
Changed, now it's "DateTimeOffset.FromUnixTimeSeconds(payload.ExpirationTimeSeconds.Value).DateTime.ToString()"